core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1640,56 +1506,55 @@ public void testUpgradeV1MetadataToV2ThroughTableProperty() {
         meta.replaceProperties(
             ImmutableMap.of(TableProperties.FORMAT_VERSION, "2", "key2", "val2"));
 
-    Assert.assertEquals(
-        "format version should be configured based on the format-version key",
-        2,
-        meta.formatVersion());
-    Assert.assertEquals(
-        "should not contain format-version but should contain new properties",
-        ImmutableMap.of("key2", "val2"),
-        meta.properties());
+    assertThat(meta.formatVersion())
+        .as("format version should be configured based on the format-version key")
+        .isEqualTo(2);
+    assertThat(meta.properties())
+        .as("should not contain format-version but should contain new properties")
+        .containsEntry("key2", "val2");
```

Comment:
this isn't doing exactly what the previous assertion did (and also doesn't do what the description is saying)

Reply:
Thanks for the comment on this. Change this part with `containsExactly` (I believe this tests if there's only one key-value pair and no format version there). If this still needs to be fixed, please let me know.